svsim/src/main/scala/vcs/Backend.scala

```diff
@@ -96,11 +96,60 @@ object Backend {
     assertionSettings:      Option[AssertionSettings] = None
   )
 
+  /** Settings for controlling VCS coverage.
+    *
+    * These options map to the `-cm` option.  Each parameter turns on a specific
+    * kind of coverage.
+    *
+    * @param line enable line coverage
+    * @param cond enable condition coverage
+    * @param fsm enable finite-state-machine coverage
+    * @param tgl enable toggle coverage
+    * @param branch enable branch coverage
+    * @param assert enable assert coverage
+    */
+  // Note: This case class is being clever and re-using the names of the
+  // elements as the names of the options.  This makes these very terse, but
+  // they then match, exactly, the documentation in the VCS manual.
+  final case class CoverageSettings(
+    line:   Boolean = false,
+    cond:   Boolean = false,
+    fsm:    Boolean = false,
+    tgl:    Boolean = false,
+    branch: Boolean = false,
+    assert: Boolean = false
+  ) {
+    def compileFlags: Seq[String] = productElementNames.zip(productIterator).flatMap {
+      case (_, false)   => None
+      case (name, true) => Some(name)
+    } match {
+      case coverageTypes if !coverageTypes.isEmpty =>
```

Comment:
It's a bit nitty but I'd just assign to a val named `coverageTypes` and then have an `if` rather than matching.

Reply:
Fixed in: https://github.com/chipsalliance/chisel/pull/4793/commits/a8b4257e1297c647a0539ab8156513a9f2fef997